src/main/java/org/folio/rest/impl/EholdingsPackagesImpl.java

```diff
@@ -120,8 +127,36 @@ public void getEholdingsPackages(String filterCustom, String q, String filterSel
   }
 
   @Override
+  @HandleValidationErrors
   public void postEholdingsPackages(String contentType, PackagePostRequest entity, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    asyncResultHandler.handle(Future.succeededFuture(Response.status(Response.Status.NOT_IMPLEMENTED).build()));
+
+    headerValidator.validate(okapiHeaders);
+    packagesPostBodyValidator.validate(entity);
+
+    PackagePost packagePost = converter.convertToPackage(entity);
+
+    MutableObject<RMAPIService> service = new MutableObject<>();
+    CompletableFuture.completedFuture(null)
+      .thenCompose(o -> configurationService.retrieveConfiguration(new OkapiData(okapiHeaders)))
+      .thenAccept(rmapiConfiguration ->
+        service.setValue(new RMAPIService(rmapiConfiguration.getCustomerId(),
+          rmapiConfiguration.getAPIKey(), rmapiConfiguration.getUrl(), vertxContext.owner())))
+      .thenCompose(vendors  ->  service.getValue().postPackage(packagePost))
```

Comment:
shall we rename 'vendors' to 'package' instead?

Reply:
'package' is one of the Java Language Keywords and variable can not have the same name. I have renamed it to 'packageById'